examples/bri-3/src/bri/transactions/agents/transactionStorage.agent.ts

```diff
@@ -0,0 +1,102 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { PrismaService } from '../../../../prisma/prisma.service';
+import { NOT_FOUND_ERR_MESSAGE } from '..//api/err.messages';
+import { Transaction } from '../models/transaction';
+
+@Injectable()
+export class TransactionStorageAgent extends PrismaService {
+  async getTransactionById(id: string): Promise<Transaction> {
+    const transactionModel = await this.transaction.findUnique({
+      where: { id: id },
+    });
+
+    if (!transactionModel) {
+      throw new NotFoundException(NOT_FOUND_ERR_MESSAGE);
+    }
+
+    return new Transaction(
+      transactionModel.id,
+      transactionModel.nonce,
+      transactionModel.workflowInstanceId,
+      transactionModel.workstepInstanceId,
+      null, // TODO: transactionModel.from once BpiAccount in the prisma schema,
+      null, // TODO: transactionModel.to once BpiAccount in the prisma schema,
+      transactionModel.payload,
+      transactionModel.signature,
+      transactionModel.status,
+    );
+  }
+
+  async getAllTransactions(): Promise<Transaction[]> {
```

Comment:
see my comments from the related RP #533

Reply:
resolved